src/decision/location_selection.jl

```diff
@@ -202,17 +205,18 @@ Note: By default, ranks are aggregated over time where `ranks` is a 3-dimensiona
 # Returns
 Frequency with which each location was selected for each rank.
 """
-function ranks_to_frequencies(ranks::NamedDimsArray, n_ranks::Int64)::NamedDimsArray
-    dn = NamedDims.dimnames(ranks)
-    freq_dims = [n for n in dn if n != :scenarios]
+function ranks_to_frequencies(ranks::YAXArray, n_ranks::Int64=length(ranks.sites))::YAXArray
+    dn = caxes(ranks)
+    freq_dims = [name(n) for n in dn if name(n) != :scenarios]
     dn_subset = vcat(freq_dims, [:ranks])
     freq_elements = vcat(
-        [1:size(ranks, n) for n in dn if n != :scenarios],
-        [1:size(ranks, :sites)],
+        [range(1, size(ranks, n), length=size(ranks, n)) for n in dn if name(n) != :scenarios],
+        [range(1, size(ranks, :sites), length=size(ranks, :sites))],
     )
+    axlist = Tuple(Dim{nm}(ran) for (nm, ran) in zip(dn_subset, freq_elements))
     mn = ([size(ranks, k) for k in freq_dims]..., size(ranks, :sites))
 
-    rank_frequencies = NamedDimsArray(zeros(mn...); zip(dn_subset, freq_elements)...)
+    rank_frequencies = YAXArray(axlist, zeros(mn...))
```

Comment:
I believe you could use the new constructor `ZeroDataCube` for this (there are a few examples at `yax_main` branch)